lib/jekyll/page.rb

```diff
@@ -36,22 +35,22 @@ class Page
     # dir  - The String path between the source and the file.
     # name - The String filename of the file.
     def initialize(site, base, dir, name)
-      @site = site
       @base = base
-      @dir  = dir
+      @dir = dir
       @name = name
-      @path = site.in_source_dir(base, dir, name)
 
-      process(name)
-      read_yaml(File.join(base, dir), name)
-
-      data.default_proc = proc do |_, key|
-        site.frontmatter_defaults.find(File.join(dir, name), type, key)
-      end
+      full_path = site.in_source_dir(base, dir, name)
+      super(full_path, {
+        :site => site,
+        :collection => site.pages
+      })
 
-      Jekyll::Hooks.trigger :pages, :post_init, self
+      process(@name)
+      read(path: full_path)
```

Comment:
@alfredxing keywords are almost literally hash arguments in Ruby, so you can do either when sending a message to the method, but when you are creating the signature it needs to be `def method(kwd: :val)`, however, like mentioned, when sending a message it can be `method(:kwd => :other_val)`